wallet/src/main/scala/org/bitcoins/wallet/models/TransactionTable.scala

```diff
@@ -0,0 +1,106 @@
+package org.bitcoins.wallet.models
+
+import org.bitcoins.core.protocol.transaction.Transaction
+import slick.jdbc.SQLiteProfile.api._
+import slick.lifted.ProvenShape
+import org.bitcoins.db.TableAutoInc
+import org.bitcoins.core.crypto.DoubleSha256DigestBE
+import org.bitcoins.db.DbRowAutoInc
+import org.bitcoins.core.protocol.script.ScriptPubKey
+import org.bitcoins.core.protocol.transaction.TransactionOutput
+
+/**
+  * Database representation of transactions
+  * relevant to our wallet.
+  */
+sealed trait TransactionDb[T <: TransactionDb[_]] extends DbRowAutoInc[T] {
+  val transaction: Transaction
+  lazy val txid: DoubleSha256DigestBE = transaction.txIdBE
+  val confirmations: Int
+
+}
+
+/** Transactions our wallet has received */
+final case class IncomingTransaction(
```

Comment:
So I think this should be renamed to `IncomingUtxo`, as this has information relevant to the utxo not necessarily the tx itself. A `IncomingTransaction` shouldn't really be concerned about a `sriptPubKey`, `outputIdx` etc